src/http/http_sig.h

```diff
@@ -285,9 +279,11 @@ namespace http
           auto k = p.substr(0, eq_pos);
           auto v = p.substr(eq_pos + 1);
 
-          // Remove inverted commas around value
-          v.remove_prefix(v.find_first_of("\"") + 1);
-          v.remove_suffix(v.size() - v.find_last_of("\""));
+          // Remove quotes around value, if present
+          if (v.size() >= 2 && v[0] == '"' && v[v.size() - 1] == '"')
+          {
+            v = v.substr(1, v.size() - 2);
```

Comment:
This is the actual fix: Only try to strip if quotes are present. Also avoid unnecessary `remove_XXXfix`, which have UB on invalid ranges, whereas substr will only throw.

Reply:
Should we error early on unmatched quotes?

Reply:
Yeah this is probably worth doing. It blocks something like `separator="`, but I think that's an edge case we don't need to support (and it remains _possible_ as `separator="""`), and should produce a more useful error when we misparse `foo="bar,baz"` as `foo="bar`, `baz"`.

Reply:
We now return an error for unbalanced quotes, and test this in `http_test`.